packages/eventual-send/src/E.js

```diff
@@ -29,8 +31,23 @@ const baseFreezableProxyHandler = {
 function EProxyHandler(x, HandledPromise) {
   return harden({
     ...baseFreezableProxyHandler,
-    get(_target, p, _receiver) {
-      return harden((...args) => HandledPromise.applyMethod(x, p, args));
+    get(_target, p, receiver) {
+      return harden(
+        {
+          [p](...args) {
```

Comment:
```suggestion
          // This function purposely uses `this` for diagnostic purposes
          // (link to explanation?)
          // In order to be `this` sensitive it is defined using concise method
          // syntax rather than as an arrow function. To ensure the function
          // is not constructable, it also avoids the `function` syntax.
          [p](...args) {
```

At some point we will bulk covert all our normal functions to arrow functions. We need to be sure that any such automatic conversion skips all functions that mention `this`.